src/services/prepareSubmissionData.ts

```diff
@@ -126,3 +126,41 @@ async function uploadAttachments(
 
   return submission
 }
+
+export function getPaymentElementValuePath(
+  elementId: string,
+  elements: FormTypes.FormElement[],
+): string[] {
+  for (const element of elements) {
+    if (element.id === elementId && element.type !== 'page') {
+      return [element.name]
+    }
+    if (element.type === 'section' && Array.isArray(element.elements)) {
+      const nestedPath = getPaymentElementValuePath(elementId, element.elements)
+
+      // Found path to element inside nested elements
+      if (nestedPath.length) {
+        return [element.name, ...nestedPath]
+      }
+    }
+  }
+  return []
+}
+
+export function getPaymentValueFromPath(
```

Comment:
From a consumption point of view, would prob be better to just export this function and rename it to `getPaymentValue()` and pass in `submission` and `elements`.  

Reply:
I'd prefer to test them separately is all

Reply:
and I figure we don't want to have an extra function exported that is used in the code, while still exporting these two for testing, because then there will be unused functions (code wise) being exported